subprojects/enterprise/src/integTest/groovy/org/gradle/internal/enterprise/core/BuildScanAutoApplyClasspathIntegrationTest.groovy

```diff
@@ -0,0 +1,142 @@
+        fixture.runtimeVersion = PLUGIN_MINIMUM_NON_DEPRECATED_VERSION
+        fixture.artifactVersion = PLUGIN_MINIMUM_NON_DEPRECATED_VERSION
+
+        file("build-src/my-plugin/build.gradle") << """
+            plugins {
+                id 'java-gradle-plugin'
+            }
+
+            repositories {
+                 maven { url '${mavenRepo.uri}' }
+            }
+
+            gradlePlugin {
+                plugins {
+                    enterprise {
+                        id = 'my.plugin'
+                        implementationClass = 'org.gradle.reproducer.MyPlugin'
+                    }
+                }
+
+                dependencies {
+                   implementation '${AutoAppliedGradleEnterprisePlugin.GROUP}:${AutoAppliedGradleEnterprisePlugin.NAME}:${fixture.artifactVersion}'
+                }
+            }
+        """
+        file("build-src/my-plugin/src/main/java/org/gradle/reproducer/MyPlugin.java") << """
+            package org.gradle.reproducer;
+
+            import org.gradle.api.Plugin;
+            import org.gradle.api.initialization.Settings;
+
+            class MyPlugin implements Plugin<Settings> {
+                @Override
+                public void apply(Settings settings) {
+                    settings.getPluginManager().apply("com.gradle.enterprise");
+                }
+            }
+        """
+
+        file("build-src/settings.gradle") << """
+            rootProject.name = 'build-src'
+
+            include 'my-plugin'
+        """
+
+        settingsFile << """
+            pluginManagement {
+                includeBuild 'build-src'
+                repositories {
+                    maven { url '${mavenRepo.uri}' }
+                }
+            }
+
+            plugins {
+                id 'my.plugin'
+            }
+        """
+
+        file('src/dummy-input.txt') << "dummy text"
+
+        buildFile << """
+            task cacheableTask {
+                def inputFile = file("src/dummy-input.txt")
+                def outputDir = file(layout.buildDirectory.dir("dummy-output"))
+                def outputFile = file(layout.buildDirectory.file("dummy-output/result.txt"))
+
+                outputs.cacheIf { true }
+
+                inputs.files(inputFile)
+                    .withPropertyName("dummy-input")
+                    .withPathSensitivity(PathSensitivity.RELATIVE)
+
+                outputs.dir(outputDir)
+                    .withPropertyName('dummy-output')
+
+                doLast {
+                    outputFile.text = inputFile.text
+                }
+            }
+        """
+    }
+
+    def "transitively applied build scan plugin disables auto-application"() {
+        when:
+        succeeds "cacheableTask", "--${BuildScanOption.LONG_OPTION}"
+
+        then:
+        fixture.appliedOnce(output)
+        fixture.assertAutoApplied(output, false)
+    }
+
+    def "task is fetched from cache when using --scan"() {
```

Comment:
```suggestion
    def "task is up-to-date when using --scan"() {
```

It won't be fetched from the build cache since the build cache is not switched on for your test.